### PR TITLE
fix(phase-14/38): coverage delta tripped on float error

### PR DESCRIPTION
**Pull request by thejesh23**
## What this PR does

T-001, the lesson's documented clean-pass scenario, was blocked by float rounding.

## Kind of change

- [x] Fix to an existing lesson

## Checklist

- [x] Code runs without errors with the listed dependencies
- [x] No comments in code files
- [x] One lesson per commit
- [x] Tested locally / code output matches what `docs/en.md` claims

## Phase / lesson

Phase 14 · 38-verification-gates

## Detail

`docs/en.md` says the gate blocks only when coverage drops **more than** 1 percentage point. T-001 ships `current=0.84 / previous=0.85` — a drop of exactly 1 pp, so it should warn. But `0.85 - 0.84 == 0.010000000000000009`, which is `> 0.01`.

The output contradicted itself:

```
task T-001: passed=False findings=1
  [block] coverage.regression: coverage dropped 1.00% (prev 85.00% -> 84.00%)
```

After:

```
task T-001: passed=True findings=1
  [warn] coverage.minor_regression: coverage dropped 1.00%
```

T-002 and T-003 are unchanged.

### CODE REVIEW

File: phases/14-agent-engineering/38-verification-gates/code/main.py [hunk 1]
```diff
@@ -127,7 +127,7 @@ def _coverage_findings(art: Artifacts, floor: float) -> list[Finding]:
     if current < floor:
         findings.append(Finding("coverage.below_floor", "block",
                                 f"coverage {current:.2%} below floor {floor:.0%}"))
-    delta = previous - current
+    delta = round(previous - current, 9)
     if delta > COVERAGE_REGRESSION_DELTA:
```
[the remark this answers is not in the document]
Comment (thejesh23) [reply phases/14-agent-engineering/38-verification-gates/code/main.py]: Valid — fixed.

Rounding the compared value did let a real regression round down to the threshold. Now comparing the raw delta and using `math.isclose` only to recognise the exactly-1pp case, so representation error decides nothing in either direction.

```
exact 1pp (T-001)            -> warn:coverage.minor_regression
just over 1pp (0.8399999996) -> block:coverage.regression
clear 2pp                    -> block:coverage.regression
no change / improved         -> no finding
```

The lesson ships no test file, so the boundary cases are exercised against `_coverage_findings` directly rather than added as tests.
